zerver/worker/queue_processors.py

```diff
@@ -481,29 +482,36 @@ def consume(self, event: Dict[str, Any]) -> None:
         self.ensure_timer()
 
     def ensure_timer(self) -> None:
-        if self.timer_event is not None:
+        if not self.lock.acquire(blocking=False):
             return
```

Comment:
What's the thinking behind having this not just use `with self.lock`?

Reply:
I added a comment explaining this - in essence, we don't want to block `consume()`, because consuming continuously to keep unconsumed events from stacking up inside rabbitmq was the point of the design of this queue in 5cec566cb9a2bff46536d87e1ec289833656ec20

Reply:
I think it's OK to block consume while we're actually doing processing; the design goal was just to ensure that we're still emptying the queue even when there's work queued by the timers?